internal/util/kubernetes/config_watcher.go

```diff
@@ -38,7 +40,7 @@ func StartConfigMapWatcher(ctx context.Context, config *rest.Config) error {
 func watchConfigMaps(ctx context.Context, client kubernetes.Interface, namespace string) {
```

Comment:
nit: the contract of this function currently makes it seem like we're watching ConfigMaps specifically in the `namespace` that's passed in as a parameter, which we're not. We could either just rename the parameter as `numaplaneNamespace` or something like that, or don't pass in `namespace` at all (the caller is calling `os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")` which could theoretically happen here)